evil-visual-mark-mode.el

```diff
@@ -109,9 +108,10 @@ This function is called on `evil-normal-state-entry-hook'."
 This function is called on the initialization of
 `evil-visual-mark-mode'"
   (evil-visual-mark-populate-overlay-alist)
-  (--each evil-visual-mark-overlay-alist
-    (evil-visual-mark-overlay-put (car it)
-                                  (cdr it))))
+  (when (evil-normal-state-p)
```

Comment:
Have you found a case where this code doesn't execute on evil-normal-state already?